### PR TITLE
add ConfigurableExtensionCustomizer

### PR DESCRIPTION
**Pull request by freshchen**
we needs to dynamically open different feature according to different scenarios.
After adding this, most default feature choose can be done through config files without writing code

### CODE REVIEW

File: pebble/src/main/java/io/pebbletemplates/pebble/extension/core/ConfigurableExtensionCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * @author freshchen
+ * @since 2024/3/2
+ */
+public class ConfigurableExtensionCustomizer {
```
Comment (ebussieres) [inline pebble/src/main/java/io/pebbletemplates/pebble/extension/core/ConfigurableExtensionCustomizer.java:25]: Maybe rename this class to DisallowExtensionCustomizerBuilder

File: pebble/src/main/java/io/pebbletemplates/pebble/extension/core/ConfigurableExtensionCustomizer.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  private Collection<String> disallowedUnaryOperatorSymbols;
+
+  private Collection<String> disallowedTestKeys;
+
+
+  public static ConfigurableExtensionCustomizer builder() {
+    return new ConfigurableExtensionCustomizer();
+  }
+
+  public ConfigurableExtensionCustomizer setDisallowedFunctionKeys(Collection<String> disallowedFunctionKeys) {
```
Comment (ebussieres) [inline pebble/src/main/java/io/pebbletemplates/pebble/extension/core/ConfigurableExtensionCustomizer.java:44]: remove set prefix from all method name of the builder

**Review by ebussieres — CHANGES_REQUESTED** [review]
Please modify documentation (docs/src/orchid/resources/wiki/guide/customize-defaults.md) accordingly too

File: pebble/src/main/java/io/pebbletemplates/pebble/extension/core/ConfigurableExtensionCustomizer.java (hunk 2 above)
Comment (freshchen) [reply pebble/src/main/java/io/pebbletemplates/pebble/extension/core/ConfigurableExtensionCustomizer.java]: good suggestion

**Pull request merged**